lib/wallet/src/Cardano/Wallet/Shelley/Transaction.hs

```diff
@@ -767,7 +767,7 @@ mkUnsignedTransaction networkId stakeCred ctx selection = do
                             , "either xpub or script when there is delegation"
                             , "action"
                             ]
-            let certs = L.nub $ votingCerts <> delegCerts
+            let certs = L.nub $ delegCerts <> votingCerts
```

Comment:
This one is needed as Registration cert needs to be FIRST...Otherwise node can protest

Reply:
I don't quite understand what's going on, because I would expect both `votingCerts` and `delegCerts` to contain the initial registration certificate. 🤔

Reply:
hmmm....I found one case where this was crucial and got node error....but it was without your refactoring. Anyway, it does not change the picture if we have `votingCerts <> delegCerts` or `delegCerts <> votingCerts`.

Reply:
> I found one case where this was crucial and got node error.

Yes, I believe you — I just don't understand why, and that makes me suspicious that there could be another logic bug in the code. 🤔

Reply:
Found it:

`joinStakePoolDelegationAction` must return `Tx.VoteRegisteringKey` instead of `Tx.Vote` in the voting action in order to for the second definition of certs to work when the key is not registered yet. I have fixed it.